paymentsheet/src/main/java/com/stripe/android/paymentsheet/ui/WalletButtonsInteractor.kt

```diff
@@ -152,12 +154,17 @@ internal class DefaultWalletButtonsInteractor(
                     ).takeIf {
                         walletsAllowedByMerchant.contains(WalletType.GooglePay)
                     }
-                    WalletType.Link -> WalletButton.Link(
-                        email = linkEmail
-                    ).takeIf {
-                        // Only show Link button if the Link verification state is resolved.
-                        linkEmbeddedState.verificationState is VerificationState.RenderButton &&
-                            walletsAllowedByMerchant.contains(WalletType.Link)
+                    WalletType.Link -> {
+                        WalletButton.Link(
+                            state = LinkButtonState.create(
+                                linkEmail = arguments.linkEmail,
+                                paymentDetails = linkAccountInfo.account?.displayablePaymentDetails
+                            )
```

Comment:
instead of passing just the email, now we build a link button state with different modes (default, email, paymentInfo)